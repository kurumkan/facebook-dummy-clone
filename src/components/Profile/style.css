.profile {
    background-color: #fff;
    border-radius: 5px;
    width: 60%;
    box-shadow: 0 2px 2px #ccc;
    margin: 10px auto;
    min-width: 300px;
    max-width: 1200px;
    text-align: center;
    padding: 20px;
}

.profile img{
    border-radius: 5px;
    max-width: 100%;
}