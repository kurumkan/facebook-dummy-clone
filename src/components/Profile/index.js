import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import DefaultImage from '../../images/default.jpg';
import Loader from 'components/Loader';
import './style.css';

class Profile extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getUser(this.props.params.username);
  }

  render() {
    const { username, firstName, lastName, imageUrl } = this.props;

    if(!username) {
      return <Loader />;
    }
    return (
      <section className="profile-section">
        <div className="profile">
          <img src={ imageUrl || DefaultImage}/>
          <div>
            <h2>{ firstName } { lastName }</h2>
          </div>
          <div>
            @{ username }
          </div>
        </div>
      </section>
    );
  }
}

Profile.propTypes = {
  getUser: PropTypes.func.isRequired
};

Profile.defaultProps = {

};

export default Profile;