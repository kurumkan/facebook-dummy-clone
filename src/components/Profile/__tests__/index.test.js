import Profile from '../index.js';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ProfileImg from 'images/profile.jpg';

describe('Alert', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <Profile
        params={{username: 'johndoe'}}
        activeUsers={['johndoe', 'joomrise']}
        firstName="John"
        gettingAUser={false}
        imageUrl={ProfileImg}
        lastName="Doe"
        username="johndoe"
        getUser={()=>{}}
      />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should render loader', () => {
    const wrapper = mount(
      <Profile
        params={{username: 'johndoe'}}
        activeUsers={['johndoe', 'joomrise']}
        firstName="John"
        gettingAUser={true}
        imageUrl={ProfileImg}
        lastName="Doe"
        username="johndoe"
        getUser={()=>{}}
      />
    );
    expect(wrapper.find('.loader-wrapper').length).toBe(1);
  });

  it('should call getUser', () => {
    const getUser = jest.fn();
    const wrapper = mount(
      <Profile
        params={{username: 'johndoe'}}
        activeUsers={['johndoe', 'joomrise']}
        firstName="John"
        gettingAUser={true}
        imageUrl={ProfileImg}
        lastName="Doe"
        username="johndoe"
        getUser={getUser}
      />
    );
    expect(getUser).toHaveBeenCalled();
  });

  it('should render loader', () => {
    const wrapper = mount(
      <Profile
        params={{username: 'johndoe'}}
        activeUsers={['johndoe', 'joomrise']}
        firstName="John"
        gettingAUser={true}
        imageUrl={ProfileImg}
        lastName="Doe"
        username="johndoe"
        getUser={()=>{}}
      />
    );
    expect(wrapper.find('.loader-wrapper').length).toBe(1);
  });

  it('should render online status', () => {
    const wrapper = mount(
      <Profile
        params={{username: 'johndoe'}}
        activeUsers={['johndoe', 'joomrise']}
        firstName="John"
        gettingAUser={false}
        imageUrl={ProfileImg}
        lastName="Doe"
        username="johndoe"
        getUser={()=>{}}
      />
    );
    expect(wrapper.find('.status').text()).toBe('online');
  });

  it('should render offline status', () => {
    const wrapper = mount(
      <Profile
        params={{username: 'johndoe'}}
        activeUsers={['joomrise']}
        firstName="John"
        gettingAUser={false}
        imageUrl={ProfileImg}
        lastName="Doe"
        username="johndoe"
        getUser={()=>{}}
      />
    );
    expect(wrapper.find('.status').text()).toBe('offline');
  });
});