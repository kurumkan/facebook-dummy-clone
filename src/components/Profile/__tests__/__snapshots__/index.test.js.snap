// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<section
  className="profile-section"
>
  <div
    className="profile"
  >
    <div
      className="image-wrapper"
      style={
        Object {
          "paddingTop": "100%",
          "width": "100%",
        }
      }
    >
      <img
        className="placeholder false"
        src="test-file-stub"
      />
      <img
        className="image false"
        onLoad={[Function]}
        src="test-file-stub"
      />
    </div>
    <div>
      <h2>
        John
         
        Doe
      </h2>
    </div>
    <div>
      <span>
        @
        johndoe
      </span>
      <span
        className="status online"
      >
        online
      </span>
    </div>
  </div>
</section>
`;
